railties/lib/rails/commands/console/console_command.rb

```diff
@@ -12,6 +12,53 @@ def filter_backtrace(bt)
       end
     end
 
+    class IRBConsole
+      def initialize
+        require "irb"
+        require "irb/completion"
+        IRB::WorkSpace.prepend(BacktraceCleaner)
+
+        if !Rails.env.local?
+          # Use env var here so users can override them with env var too
+          ENV["IRB_USE_AUTOCOMPLETE"] ||= "false"
+        end
+      end
+
+      def name
+        "IRB"
+      end
+
+      def start
+        IRB.setup(nil)
+
+        env = colorized_env
+
+        IRB.conf[:PROMPT][:RAILS_PROMPT] = {
+          PROMPT_I: "#{env}:%03n> ",
+          PROMPT_S: "#{env}:%03n%l ",
+          PROMPT_C: "#{env}:%03n* ",
+          RETURN: "=> %s\n"
+        }
+
+        # Respect user's choice of prompt mode.
+        IRB.conf[:PROMPT_MODE] = :RAILS_PROMPT if IRB.conf[:PROMPT_MODE] == :DEFAULT
+        IRB::Irb.new.run(IRB.conf)
+      end
+
+      def colorized_env
+        case Rails.env
+        when "development"
+          IRB::Color.colorize("dev", [:GREEN])
+        when "test"
+          IRB::Color.colorize("test", [:GREEN])
+        when "production"
+          IRB::Color.colorize("prod", [:RED])
```

Comment:
Shouldn't this be production?

Reply:
seems `prod` is intended
![image](https://github.com/rails/rails/assets/193936/ae47bb16-97db-48d1-92eb-f3533e9fdc1b)

Reply:
The original discussion in #50770 decided to use abbreviations for development and production environments.

Reply:
Yeah, same for dev, it should be development I think